src/classes/actions-cache/download.ts

```diff
@@ -0,0 +1,65 @@
+import {createHttpClient, getCacheApiUrl} from './http-client';
+import {retryTypedResponse} from './retry';
+import {isSuccessStatusCode} from './http-responses';
+import {HttpClient} from '@actions/http-client';
+import {downloadCacheHttpClient} from '@actions/cache/lib/internal/downloadUtils';
+import * as core from '@actions/core';
+
+interface ArtifactCacheEntry {
+  cacheKey?: string;
+  scope?: string;
+  cacheVersion?: string;
+  creationTime?: string;
+  archiveLocation?: string;
+}
+const getCacheArchiveUrl = async (
+  httpClient: HttpClient,
+  cacheKey: string,
+  cacheVersion: string
+): Promise<string | null> => {
+  // TODO: should work with delete?
+  const resource = `cache?keys=${cacheKey}&version=${cacheVersion}`;
+
+  const response = await retryTypedResponse('getCacheEntry', async () =>
+    httpClient.getJson<ArtifactCacheEntry>(getCacheApiUrl(resource))
+  );
+  // Cache not found
+  if (response.statusCode === 204) {
+    core.debug(
+      `There's no cache with key ${cacheKey} & version=${cacheVersion}`
+    );
+    // List cache for primary key only if cache miss occurs
+    return null;
+  }
+  if (!isSuccessStatusCode(response.statusCode)) {
+    throw new Error(`Cache service responded with ${response.statusCode}`);
+  }
+
+  const cacheResult = response.result;
+  core.debug(`getCacheEntry response is:\n${JSON.stringify(cacheResult)}`);
+  const cacheDownloadUrl = cacheResult?.archiveLocation;
+  if (!cacheDownloadUrl) {
+    // Cache archiveLocation not found. This should never happen, and hence bail out.
+    throw new Error('Cache not found.');
+  }
+  return cacheDownloadUrl;
+};
+
+export const downloadFileFromActionCache = async (
```

Comment:
There are 2 reasons to keep it:

 It is a brand name of the service used to persist the state, it is not "a some cache" and using a word cache makes a question "how a caching (usually temporary) does relate to storing (usually permanent)" 
 
 I am not 100% sure at the moment the actions cache is the perfect solution. It is possible some day it will be decided to replace it with come cloud provider (say 90 days lifetime will be not enough)